test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java

```diff
@@ -65,25 +60,41 @@ public static ByteString parsedToByteString(long n) {
 	}
 
 	public static byte[] asAddress(final TokenID id) {
-		return asSolidityAddress((int) id.getShardNum(), id.getRealmNum(), id.getTokenNum());
+		return asUtilAddress((int) id.getShardNum(), id.getRealmNum(), id.getTokenNum());
 	}
 
 	public static byte[] asAddress(final AccountID id) {
-		return asSolidityAddress((int) id.getShardNum(), id.getRealmNum(), id.getAccountNum());
+		return asUtilAddress((int) id.getShardNum(), id.getRealmNum(), id.getAccountNum());
 	}
 
 	public static byte[] asAddress(final ContractID id) {
-		return asSolidityAddress((int) id.getShardNum(), id.getRealmNum(), id.getContractNum());
+		return asUtilAddress((int) id.getShardNum(), id.getRealmNum(), id.getContractNum());
+	}
+
+	public static com.esaulpaugh.headlong.abi.Address asHeadlongAddress(byte[] address) {
+		final var besuAddress = org.hyperledger.besu.datatypes.Address.wrap(Bytes.wrap(address));
+		return convertBesuAddressToHeadlongAddress(besuAddress);
+	}
+
+	static com.esaulpaugh.headlong.abi.Address convertBesuAddressToHeadlongAddress(final org.hyperledger.besu.datatypes.Address address) {
+		return com.esaulpaugh.headlong.abi.Address.wrap(
+				com.esaulpaugh.headlong.abi.Address.toChecksumAddress(address.toUnsignedBigInteger()));
+	}
+
+	public static com.esaulpaugh.headlong.abi.Address convertAliasToHeadlongAddress(String address) {
+		var result = Bytes.fromHexStringLenient(address).toArray();
+		var besuAddress = org.hyperledger.besu.datatypes.Address.wrap(Bytes.wrap(result));
+		return convertBesuAddressToHeadlongAddress(besuAddress);
 	}
 
-	public static byte[] asSolidityAddress(final int shard, final long realm, final long num) {
-		final byte[] solidityAddress = new byte[20];
+	public static byte[] asUtilAddress(final int shard, final long realm, final long num) {
```

Comment:
Why should we rename this method? We don't have an address of type "Util". I think the name is misleading.